backend/test_firestore.go

```diff
@@ -9,43 +9,46 @@ import (
 	"github.com/ubclaunchpad/when3meet/data/clients/firebase"
 )
 
-//TODO: Discuss error handling policy / approach  with team
 func CreateUser(w http.ResponseWriter, r *http.Request) {
 	// return json
 	w.Header().Set("Content-type", "application/json")
 
 	// unmarshal the JSON in the request to the data struct
 	// to access the key-value pairs
-	var exampleUser firebase.User
-	err := json.NewDecoder(r.Body).Decode(&exampleUser)
+	var user firebase.User
+	err := json.NewDecoder(r.Body).Decode(&user)
 	if err != nil {
-		w.WriteHeader(http.StatusInternalServerError)
-		log.Fatalf("failed to decode the request: %v", err)
+		log.Warnf("Failed to decode the request: %v", err)
+		http.Error(w,"Failed to decode the provided user",http.StatusBadRequest)
+		return
 	}
 
-	err = exampleUser.Write()
+	log.Printf("Creating user with data: %+v", user)
+	err = user.Write()
 	if err != nil {
-		log.Fatalf("failed to add user : %v", err)
+		log.Warnf("Failed to add user : %v", err)
+		http.Error(w, "Something went wrong creating the user", http.StatusInternalServerError)
+		return
 	}
 
 	w.WriteHeader(http.StatusCreated)
 	// marshal the data struct to JSON to send as response
-	json.NewEncoder(w).Encode(exampleUser)
+	json.NewEncoder(w).Encode(user)
 }
 
 func CreateEvent(w http.ResponseWriter, r *http.Request) {
 	event := &firebase.Event{}
 	err := json.NewDecoder(r.Body).Decode(&event)
 	if err != nil {
-		w.WriteHeader(http.StatusBadRequest)
 		log.Warnf("Failed to decode the request: %v", err)
+		http.Error(w,"Failed to decode the provided event", http.StatusBadRequest)
 		return
 	}
-	log.Infof("Creating event with data: %+v", event)
+	log.Printf("Creating event with data: %+v", event)
```

Comment:
why print instead of info?

Reply:
Good point I think that they both log to INFO level but using PrintF will log even if info level logs are not enabled. For our sake though, we should probably be using InfoF just for more control on our end